src/compiler/types.ts

```diff
@@ -6750,6 +6752,7 @@ namespace ts {
         AssertTypeScript = ContainsTypeScript,
         AssertJsx = ContainsJsx,
         AssertESNext = ContainsESNext,
+        AssertES2022 = ContainsES2022,

```

Comment:
If ES2022 does not contain new syntax, should we skip `ContainsES2022` flag (or set `ContainsES2022` = `ContainsES2021`) to save the bitflag?

Reply:
There are (notably class private fields), but just not here yet AFAIK.